jobsdb/jobsdb.go

```diff
@@ -1128,21 +1131,31 @@ func (jd *HandleT) checkIfMigrateDS(ds dataSetT) (
 
 	recordsLeft = totalCount - delCount
 
-	// If records are newer than what is required. One example use case is
-	// gateway DB where records are kept to dedup
-
-	var minCreatedAt, maxCreatedAt time.Time
-	sqlStatement = fmt.Sprintf(`SELECT MIN(created_at), MAX(created_at) from %q`, ds.JobTable)
-	row = jd.dbHandle.QueryRow(sqlStatement)
-	err = row.Scan(&minCreatedAt, &maxCreatedAt)
-	jd.assertError(err)
+	if jd.MinDSRetentionPeriod > 0 {
+		var maxCreatedAt time.Time
+		sqlStatement = fmt.Sprintf(`SELECT MAX(created_at) from %q`, ds.JobTable)
+		row = jd.dbHandle.QueryRow(sqlStatement)
+		err = row.Scan(&maxCreatedAt)
+		jd.assertError(err)
 
-	if jd.MinDSRetentionPeriod > 0 && time.Since(maxCreatedAt) < jd.MinDSRetentionPeriod {
-		return false, false, recordsLeft
+		if time.Since(maxCreatedAt) < jd.MinDSRetentionPeriod {
+			return false, false, recordsLeft
+		}
 	}
 
-	if jd.MaxDSRetentionPeriod > 0 && time.Since(minCreatedAt) > jd.MaxDSRetentionPeriod {
-		return true, false, recordsLeft
+	if jd.MaxDSRetentionPeriod > 0 {
+		var terminalJobsExist sql.NullBool
```

Comment:
is there a case for an exists clause to return null?

Reply:
No. Used NullBool and .Valid check as a safe thing.

Reply:
I don't believe this is needed though

Reply:
removed